content/markdown/02-working-with-data/02-07-dates.Rmd

```diff
@@ -0,0 +1,108 @@
+
+```{r}
+head(df)
+```
+
+So we already have a date column of the special data type `date`. This in fact makes our life easy, as it's always easier to extract information from a pre-formatted date column. 
+
+The best tool in our toolbox for this type of task is the `lubridate` package. This package has a ton of great functions that let us work with dates more easily. Let's test it out. Say we want to make a vector that has just all the years extracted from our date column. We could do that very easily like this with the `lubridate::year()` function:
+
+```{r}
+library(lubridate)
+year <- lubridate::year(df$date) 
+
+# print the first ten entries
+year[1:10]
+```
+
+And we see that this worked! So the way we probably use this the most often is to make new columns in a dataframe. Let's go ahead and make a new column each for year, month, week, and day in our `df` dataframe:
+
+```{r}
+df %>%
+  # since we want lubridate to work with each row individually, use rowwise()
+  dplyr::rowwise() %>% 
+  # we use mutate() to make a new column
+  dplyr::mutate(
+    year = lubridate::year(date),
+    month = lubridate::month(date),
+    week = lubridate::week(date), 
+    day = lubridate::day(date)
+  )
+```
+
+And here we can see the new columns have been made for us! 
+
+## Forming Dates
+
+To start thinking about forming dates, we'll use some fake data to make our lives easier. We can imagine the opposite scenario to above, we have some entries for, let's say, year and month, but no full date. 
+
+This brings about a somewhat more challenging problem as there's a decision point that needs to be executed -- what day should we default to? This is a question that deserves careful consideration for each problem that arises and there is no one-size-fits-all solution. However, assuming you have decided that there is a simple assumption you can make (e.g. you will assume the data were collected on the first of the month), we can use this to make a new `date` column from our existing data. 
+
+Let's generate some fake data to work with: 
+
+```{r}
+df = data.frame(
+  year = sample(c(2010:2020), replace = TRUE, 200),
+  month = sample(c(1:12), replace = TRUE, 200),
+  # we'll make a set of fake sampled data here
+  observation = sample(c(12.5:16.6), replace = TRUE, 200)
+)
+```
+
+The first thing to do is inpute our decided day values of the first day of each month. That's easy enough: 
+
+```{r}
+df = df %>% 
+  dplyr::mutate(
+    day = 1
+  )
+df
+```
+
+Great, we have the info we need. Now, we can go ahead and make a `date` column by combining our three other variables together using thye `lubrdiate::make_date()` function:
+
+```{r}
+df = df %>% 
+  dplyr::rowwise() %>% 
+  dplyr::mutate(
+    date = lubridate::make_date(year, month, day)
+  )
+df
+```
+
+Great, we can visually check this by looking across the first few rows and we see our function worked as it should.
+
+Now you know how to move from dates to components and vice versa!
```

Comment:
So, for some reason the PDF link below was most of the content from above, though not all the text and the end showed me 158 rows of the arctic dataset.